samples-generator/lib/createBatchTableHierarchy.ts

```diff
@@ -142,16 +160,65 @@ function createBatchTableHierarchy(options) {
             }
         }
         var batchedMesh = Mesh.batch(clonedMeshes);
+
+        if (use3dTilesNext || useGlb) {
+            return createGltf({
+                mesh: batchedMesh
+            });
+        }
+
         return createGlb({
             mesh : batchedMesh
         });
-    }).then(function(glb) {
+    }).then(function(result) {
+        if (use3dTilesNext || useGlb) {
+            if (defined(batchTableJson)) {
+
+                // add human readable batch table data
+                if (defined(batchTableJson) && Object.keys(batchTableJson).length > 0) {
+                    var batchTableJsonPruned = {
+                        area: batchTableJson.area,
+                        height: batchTableJson.height
+                    };
+
+                    result = createFeatureMetadataExtension(
+                        result, 
+                        batchTableJsonPruned, 
+                        batchTableBinary
+                    ) as Gltf;
+
+                    const hierarchy = 
+                        batchTableJson.tilesNextHierarchy;
+
+                    if (defined(hierarchy)) {
+                        addHierarchyToGltf(hierarchy, result, batchTableBinary);
+                    }
+                }
+            }
+
+            if (!useGlb) {
+                return Promise.all([
+                    saveJson(tilePath, result, options.prettyJson, gzip),
+                    saveJson(tilesetJsonPath, tilesetJson, options.prettyJson, gzip)
+                ]);
+            } else {
+                return Promise.all([
+                    gltfToGlb(result, gltfConversionOptions).then(function(out) {
+                        saveBinary(tilePath, out.glb, gzip);
```

Comment:
Is `return` still needed?

```suggestion
                        return saveBinary(tilePath, out.glb, gzip);
```